controllers/managedocs_controller.go

```diff
@@ -1564,17 +1568,43 @@ func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
 }
 
 func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
-	// Fix for non-converged deployment type will be provided in upcoming PR
-	if r.DeploymentType != convergedDeploymentType {
-		r.Log.Info("Non converged deployment, skipping reconcile for egress network policy")
-		return nil
-	}
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
 		if err := r.own(r.egressNetworkPolicy); err != nil {
 			return err
 		}
 		desired := templates.EgressNetworkPolicyTemplate.DeepCopy()
 
+		if r.DeploymentType != convergedDeploymentType {
+			// Get the aws config map
+			awsConfigMap := corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      aws.DataConfigMapName,
+					Namespace: r.namespace,
+				},
+			}
+			if err := r.get(&awsConfigMap); err != nil {
+				return fmt.Errorf("Unable to get AWS ConfigMap: %v", err)
+			}
+
+			// Get the machine cidr
+			vpcCIDR, ok := awsConfigMap.Data[aws.CidrKey]
+			if !ok {
+				return fmt.Errorf("Unable to determine machine CIDR from AWS ConfigMap")
+			}
+			// Allow egress traffic to that cidr range
+			vpcEgressRule := openshiftv1.EgressNetworkPolicyRule{
+				Type: openshiftv1.EgressNetworkPolicyRuleAllow,
+				To: openshiftv1.EgressNetworkPolicyPeer{
+					CIDRSelector: vpcCIDR,
+				},
+			}
+
+			desired.Spec.Egress = append(
+				[]openshiftv1.EgressNetworkPolicyRule{
+					vpcEgressRule,
+				}, desired.Spec.Egress...)
```

Comment:
What is the advantage of doing it the way you suggest? What is the issue appending the rule?

Reply:
The same reasons we do everything in templates. 
The entire role can be described as template that just needs to be filled and the code part becomes
 more concise. 
 So this:
 ```
vpcEgressRule := openshiftv1.EgressNetworkPolicyRule{
	Type: openshiftv1.EgressNetworkPolicyRuleAllow,
	To: openshiftv1.EgressNetworkPolicyPeer{
		CIDRSelector: vpcCIDR,
	},
}

desired.Spec.Egress = append(
	[]openshiftv1.EgressNetworkPolicyRule{
		vpcEgressRule,
	}, desired.Spec.Egress...)
 ```
becomes:
```
desired.Spec.Egress[0].To.CIDRSelector: vpcCIDER
```

Reply:
Can you please elaborate?

Reply:
Sure: If you add the EgressNetworkPolicyRule to the template, you need to leave the value for CIDRSelector as "". That's fine for the non-converged cases since it will get mutated to the vpcCIDR once its determined, but for the converged case it will remain blank. How the EgressNetworkPolicyRule is handled when the CIDRSelector is blank is not defined behavior (At least from what I can tell in the [documentation](https://docs.openshift.com/container-platform/4.10/networking/openshift_sdn/configuring-egress-firewall.html)). 

It would be more accurate to say that it _may_ break converged, but I think its safe to go with the assumption that the undefined behavior will break it as opposed to testing that it doesn't, and then having to make sure that every update to openshift sdn doesn't change that behavior.

Reply:
I can use a template if you want, I just think it would be better to use a separate template for non-converged cases then.

Reply:
I see. But we do have a way to separate code paths in our code base. so why can't we just mutate the template in accordance with the deployment type? 
```
r.DeploymentType == convergedDeploymentType {
    desired.Spec.Egress[0] = nil
} else {
    desired.Spec.Egress[0].To.CIDRSelector: vpcCIDER
}
```
But I guess my suggestion is suffering from the problem of relying on a constant placement in the array (position 0), and there is no way to query a rule (in the template) by some sort of identity. So for now I think your suggestion is better with a small request for change. Instead of 


I see. But we do have a way to separate code paths in our code base. so why can't we just mutate the template in accordance with the deployment type? 
```
r.DeploymentType == convergedDeploymentType {
    desired.Spec.Egress[0] = nil
} else {
    desired.Spec.Egress[0].To.CIDRSelector: vpcCIDER
}
```
But I guess my suggestion is suffering from the problem of relying on a constant placement in the array (position 0), and there is no way to query a rule (in the template) by some sort of identity. So, for now, I think your suggestion is better with a small request for change. Instead of adding the new rule at the start of the array, let's append at the end:
```
desired.Spec.Egress = append(
	desired.Spec.Egress,
	openshiftv1.EgressNetworkPolicyRule{
		Type: openshiftv1.EgressNetworkPolicyRuleAllow,
		To: openshiftv1.EgressNetworkPolicyPeer{
			CIDRSelector: vpcCIDR,
		},
	},
)
```
As it results in a more concise and readable code.
Unless there is a concrete reason for it to be the first rule.

Reply:
There is a reason why I inserted the rule in the front instead of appending it: [The order of the rules matter](https://docs.openshift.com/container-platform/4.10/networking/openshift_sdn/configuring-egress-firewall.html#policy-rule-order_openshift-sdn-egress-firewall). This rule must come before the EgressNetworkPolicyRuleDeny rule. Inserting the rule in the front ensures that the vpc policy comes before it.

I can add a comment explaining this

Reply:
Please do so, It will clarify this bit for a future reader.